### ISSUE TITLE
[BUG] Drag & drop failing on Obsidian 1.5.3 (Installer 1.5.3)

### ISSUE THREAD

**Issue by shawnyeager** [vslinko/obsidian-outliner#506]
**Describe the bug**
I'm unable to drag & drop using both Default and Minimal themes.

**To Reproduce**

1. Grab icon appears while hovered over item
2. Click to grab
3. Attempt to drag

**Expected behavior**
The item moves with the cursor

**Environment (please complete the following information):**
 - OS: macOS 14.3
 - Obsidian Version: Version 1.5.3 (Installer 1.5.3)
 - Plugin Version: 4.8.0

**Comment by imdadkadir** [comment]
https://forum.obsidian.md/t/fix-for-drag-and-drop-functionality-in-outliner-plugin/76425?u=kadz

Let me know if this helps!

**Comment by charlie17** [comment]
I have discovered something by accident - DnD seems to work properly when all outline nodes in the file are expanded. As soon as some are collapsed it stops working. See if this happens for you?

**Comment by charlie17** [comment]
Yes, I tried it in some different files and found I couldn't repeat what I found before. There is something going on where it sometimes works, but I can't crack the case on exactly what that is. Frustrating as DnD is very helpful on the few occasions where it does work

**Comment by HWiwoiiii** [comment]
same here

**Comment by TonyDobbs** [comment]
I'm experiencing the same issue. Has anyone tried the fix recommended by @imdadkadir ?

**Comment by charlie17** [comment]
> I'm experiencing the same issue. Has anyone tried the fix recommended by @imdadkadir ?

I saw that but haevn't tried it only because DnD doesn't work even in cases where I don't have any front matter

**Comment by Yaozhuwa** [comment]
> https://forum.obsidian.md/t/fix-for-drag-and-drop-functionality-in-outliner-plugin/76425?u=kadz
> 
> Let me know if this helps!如果有帮助，请告诉我！

It work for me~

**Comment by TonyDobbs** [comment]
> > I'm experiencing the same issue. Has anyone tried the fix recommended by @imdadkadir ?
> 
> I saw that but haevn't tried it only because DnD doesn't work even in cases where I don't have any front matter

Sorry about that @charlie17 I wasn't looking carefully. I just submitted a PR to resolve this issue.

**Comment by kvikindi** [comment]
I'm on v1.5.11, and the only way to get DnD to work, is if "Properties in Document" is set to "Source" (under Options - Editor - Display).

**Comment by candideu** [comment]
Same issue for me

**Comment by TonyDobbs** [comment]
> Same issue for me

Hi @candideu  did you try my patch above?

**Comment by TonyDobbs** [comment]
> @TonyDobbs I did. Based on [your commit](https://github.com/vslinko/obsidian-outliner/pull/515/files), I went to to `[my vault]\.obsidian\plugins\obsidian-outliner`, opened main.js in a text editor, found the following line (L1310):
> 
> ```js
>       v.top = view.coordsAtPos(linePos, -1).top;
> ```
> 
> and replaced it with this:
> 
> ```js
>       const coords = view.coordsAtPos(linePos, -1);
>       const vTop = coords ? coords.top : 0;
>       v.top = vTop;
> ```
> 
> Then I restarted Obsidian. Unfortunately, it's still not working for me...

@candideu Ah I think the issue is that if you only update main.js you won't see the changes, because it's using a minified version. You will need to update the code in that file as well.

**Comment by candideu** [comment]
> You will need to update the code in that file as well.

@TonyDobbs Sorry in which file?


I ended up combining your changes with the [changes described in the forum post](https://forum.obsidian.md/t/fix-for-drag-and-drop-functionality-in-outliner-plugin/76425) that was shared [earlier in this thread](https://github.com/vslinko/obsidian-outliner/issues/506#issuecomment-1929304229) in `main.js`, and it looks like that worked.

**Comment by TonyDobbs** [comment]
> > You will need to update the code in that file as well.
> 
> @TonyDobbs Sorry in which file?
> 
> I ended up combining your changes with the [changes described in the forum post](https://forum.obsidian.md/t/fix-for-drag-and-drop-functionality-in-outliner-plugin/76425) that was shared [earlier in this thread](https://github.com/vslinko/obsidian-outliner/issues/506#issuecomment-1929304229) in `main.js`, and it looks like that worked.

Hi @candideu My apologies, main.js was the correct file. I was mistaken. Based on your reply, it seems that you may have been using frontmatter. So there were 2 bugs to patch not one. I'm glad to hear you got it working! Hopefully both patches make it into the next release.
